pkg/proc/gdbserial/gdbserver_conn.go

```diff
@@ -787,33 +834,31 @@ func (conn *gdbConn) parseStopPacket(resp []byte, threadID string, tu *threadUpd
 				// mach exception data (debugserver extension)
 				d, _ := strconv.ParseUint(string(value), 16, 64)
 				medata = append(medata, d)
-			}
-		}
-
-		// Debugserver does not report watchpoint stops in the standard way preferring
-		// instead the semi-undocumented metype/medata keys.
-		// These values also have different meanings depending on the CPU architecture.
-		switch conn.goarch {
-		case "amd64":
-			if metype == _EXC_BREAKPOINT && len(medata) >= 2 && medata[0] == _EXC_I386_SGL {
-				sp.watchAddr = medata[1] // this should be zero if this is really a single step stop and non-zero for watchpoints
-			}
-		case "arm64":
-			if metype == _EXC_BREAKPOINT && len(medata) >= 2 && (medata[0] == _EXC_ARM_DA_DEBUG || medata[0] == _EXC_ARM_BREAKPOINT) {
-				// The arm64 specification allows for up to 16 debug registers.
-				// The registers are zero indexed, thus a value less than 16 will
-				// be a hardware breakpoint register index.
-				// See: https://developer.arm.com/documentation/102120/0101/Debug-exceptions
-				// TODO(deparker): we can ask debugserver for the number of hardware breakpoints
-				// directly.
-				if medata[1] < 16 {
-					sp.watchReg = int(medata[1])
-				} else {
-					sp.watchAddr = medata[1]
+			case "jstopinfo":
+				jstopinfo, err := hex.DecodeString(string(value))
+				if err != nil {
+					return false, stopPacket{}, fmt.Errorf("malformed stop packet: %s (wrong jstopinfo)", string(resp))
+				}
+				parsedJstopInfo := []jsonStopPacket{}
+				if err := json.Unmarshal(jstopinfo, &parsedJstopInfo); err != nil {
+					return false, stopPacket{}, fmt.Errorf("malformed stop packet: %s (wrong jstopinfo)", string(resp))
+				}
+				sp.jstopInfo = make(map[int]stopPacket)
+				for _, jsp := range parsedJstopInfo {
+					threadID := fmt.Sprintf("%d", jsp.Tid)
+					newStopPacket := stopPacket{
+						threadID: threadID,
+						reason:   jsp.Reason,
+						sig:      uint8(jsp.Signal),
+					}
+					parseMachException(&newStopPacket, jsp.Metype, jsp.Medata)
+					sp.jstopInfo[jsp.Tid] = newStopPacket
 				}
 			}
 		}
 
+		parseMachException(&sp, metype, medata)
```

Comment:
Why not leave this inlined?

Reply:
It's being used in the parsing of the `jstopinfo` section as well. I could remove that, we're not really using that information as it's somewhat incomplete in my testing (doesn't include signal, but that *may* only hold true for the main thread mentioned in the `T` packet).

Reply:
I see, I think it's fine to leave it here.